src/middleware.ts

```diff
@@ -0,0 +1,162 @@
+import {NextRequest, NextResponse} from 'next/server'
+import {ACCESS_TOKEN_KEY, EGGHEAD_USER_COOKIE_KEY} from './config'
+import {loadCio} from './lib/customer'
+import {loadUser} from './lib/users'
+
+const PUBLIC_FILE = /\.(.*)$/
+
+const CIO_COOKIE_KEY = 'cio_id'
+
+// Supports both a single string value or an array of matchers
+export const config = {
+  matcher: ['/', '/pricing'],
+}
+
+/**
+ * with this approach, logged in users can be shown
+ * '/' and anon users '/signup' IN PLACE
+ *
+ * we don't have a server cookie to get member status, but that
+ * should be on the short list. That would make this even better
+ * because we could send them into a more specific /learn route
+ * and we could start personalizing that a bit more using various
+ * cookies and such.
+ *
+ * This looks a lot like the i18n example:
+ * https://github.com/vercel/examples/tree/main/edge-functions/i18n
+ *
+ * an the split testing example that puts them in a bucket:
+ * https://github.com/vercel/examples/edge-functions/ab-testing-simple
+ *
+ */
+export async function middleware(req: NextRequest) {
+  let response = NextResponse.next()
+  // think favicon etc
+  if (PUBLIC_FILE.test(req.nextUrl.pathname)) return response
+
+  const url = req.nextUrl.clone()
+
+  const cioId =
+    req.cookies.get(CIO_COOKIE_KEY) ||
+    req.nextUrl.searchParams.get(CIO_COOKIE_KEY)
+
+  const eggheadAccessToken = req.cookies.get(ACCESS_TOKEN_KEY)
+
+  // if there's a cookie or a token they are logged in
+  let status = cioId || eggheadAccessToken ? 'identified' : 'anon'
+
+  // if there's a token, load the user (use cookie cached user if available)
+  const user =
+    eggheadAccessToken &&
+    (await loadUser(
+      eggheadAccessToken,
+      req.cookies.get(EGGHEAD_USER_COOKIE_KEY),
+    ))
```

Comment:
previously we weren't loading the egghead user and I noticed that it had issues if for some reason the CIO user wasn't updated etc so this handles that. if the user is already cookied it will use that.